src/components/date-picker/date-picker.ts

```diff
@@ -0,0 +1,771 @@
+ * @slot calendar-icon - Renders the icon/content for the calendar picker.
+ * @slot calendar-icon-open - Renders the icon/content for the picker in open state.
+ * @slot actions - Renders content in the action part of the picker in open state.
+ *
+ * @fires igcOpening - Emitted just before the calendar dropdown is shown.
+ * @fires igcOpened - Emitted after the calendar dropdown is shown.
+ * @fires igcClosing - Emitted just before the calendar dropdown is hidden.
+ * @fires igcClosed - Emitted after the calendar dropdown is hidden.
+ * @fires igcChange - Emitted when the user modifies and commits the elements's value.
+ * @fires igcInput - Emitted when when the user types in the element.
+ *
+ * @csspart label - The label wrapper that renders content above the target input.
+ * @csspart container - The main wrapper that holds all main input elements.
+ * @csspart input - The native input element.
+ * @csspart prefix - The prefix wrapper.
+ * @csspart suffix - The suffix wrapper.
+ * @csspart calendar-icon - The calendar icon wrapper for closed state.
+ * @csspart calendar-icon-open - The calendar icon wrapper for opened state.
+ * @csspart clear-icon - The clear icon wrapper.
+ * @csspart actions - The actions wrapper.
+ * @csspart helper-text - The helper-text wrapper that renders content below the target input.
+ * @csspart header - The calendar header element.
+ * @csspart header-title - The calendar header title element.
+ * @csspart header-date - The calendar header date element.
+ * @csspart calendar-content - The calendar content element which contains the views and navigation elements.
+ * @csspart navigation - The calendar navigation container element.
+ * @csspart months-navigation - The calendar months navigation button element.
+ * @csspart years-navigation - The calendar years navigation button element.
+ * @csspart years-range - The calendar years range element.
+ * @csspart navigation-buttons - The calendar navigation buttons container.
+ * @csspart navigation-button - The calendar previous/next navigation button.
+ * @csspart days-view-container - The calendar days view container element.
+ * @csspart days-view - The calendar days view element.
+ * @csspart months-view - The calendar months view element.
+ * @csspart years-view - The calendar years view element.
+ * @csspart days-row - The calendar days row element.
+ * @csspart calendar-label - The calendar week header label element.
+ * @csspart week-number - The calendar week number element.
+ * @csspart week-number-inner - The calendar week number inner element.
+ * @csspart date - The calendar date element.
+ * @csspart date-inner - The calendar date inner element.
+ * @csspart first - The calendar first selected date element in range selection.
+ * @csspart last - The calendar last selected date element in range selection.
+ * @csspart inactive - The calendar inactive date element.
+ * @csspart hidden - The calendar hidden date element.
```

Comment:
Well, there are documented [here](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/scheduling/calendar#styling) and [here](https://github.com/IgniteUI/igniteui-webcomponents/wiki/Calendar-Specification#css-parts) so that is that.

Reply:
Ah, just not here:
https://github.com/IgniteUI/igniteui-webcomponents/blob/41187068318c44dcc96a592c95c561e805262f16/src/components/calendar/calendar.ts#L46-L61